tests/integration/stepfunctions/test_stepfunctions_v2.py

```diff
@@ -0,0 +1,797 @@
+    )
+    policy = iam_client.create_policy(
+        PolicyDocument='{"Version": "2012-10-17", "Statement": {"Action": "sns:createTopic", "Effect": "Allow", "Resource": "*"}}',
+        PolicyName=policy_name,
+    )
+    iam_client.attach_role_policy(
+        RoleName=role["Role"]["RoleName"], PolicyArn=policy["Policy"]["Arn"]
+    )
+
+    result = stepfunctions_client.create_state_machine(
+        name=name,
+        definition=json.dumps(statemachine_definition),
+        roleArn=role["Role"]["Arn"],
+    )
+    machine_arn = result["stateMachineArn"]
+
+    try:
+        result = stepfunctions_client.list_state_machines()["stateMachines"]
+        assert len(result) > 0
+        assert len([sm for sm in result if sm["name"] == name]) == 1
+
+        def assert_execution_success(executionArn: str):
+            def _assert_execution_success():
+                status = stepfunctions_client.describe_execution(executionArn=executionArn)[
+                    "status"
+                ]
+                if status == "FAILED":
+                    raise ShortCircuitWaitException("Statemachine execution failed")
+                else:
+                    return status == "SUCCEEDED"
+
+            return _assert_execution_success
+
+        def _retry_execution():
+            # start state machine execution
+            # AWS initially straight up fails until the permissions seem to take effect
+            # so we wait until the statemachine is at least running
+            result = stepfunctions_client.start_execution(
+                stateMachineArn=machine_arn, input='{"Name": "' f"{topic_name}" '"}'
+            )
+            assert wait_until(assert_execution_success(result["executionArn"]))
+            describe_result = stepfunctions_client.describe_execution(
+                executionArn=result["executionArn"]
+            )
+            output = describe_result["output"]
+            assert topic_name in output
+
+            # TODO: implement stepfunction's 'describe_state_machine_for_execution'.
+            # result = stepfunctions_client.describe_state_machine_for_execution(
+            #     executionArn=result["executionArn"]
+            # )
+            # assert result["stateMachineArn"] == machine_arn
+
+            topic_arn = json.loads(describe_result["output"])["TopicArn"]
+            topics = sns_client.list_topics()
+            assert topic_arn in [t["TopicArn"] for t in topics["Topics"]]
+            sns_client.delete_topic(TopicArn=topic_arn)
+            return True
+
+        assert wait_until(_retry_execution, max_retries=3, strategy="linear", wait=3.0)
+
+    finally:
+        iam_client.detach_role_policy(RoleName=role_name, PolicyArn=policy["Policy"]["Arn"])
+        iam_client.delete_role(RoleName=role_name)
+        iam_client.delete_policy(PolicyArn=policy["Policy"]["Arn"])
+        stepfunctions_client.delete_state_machine(stateMachineArn=machine_arn)
```

Comment:
Not blocking a merge, but we should go over the tests in here soon and try to "modernize" them a bit to follow our current testing conventions :+1: